monai/transforms/io/dictionary.py

```diff
@@ -106,4 +110,91 @@ def __call__(self, data, reader: Optional[ImageReader] = None):
+
+            - NIfTI files {``"bilinear"``, ``"nearest"``}
+                Interpolation mode to calculate output values.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#grid-sample
+            - PNG files {``"nearest"``, ``"linear"``, ``"bilinear"``, ``"bicubic"``, ``"trilinear"``, ``"area"``}
+                The interpolation mode.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#interpolate
+
+        padding_mode: This option is used when ``resample = True``. Defaults to ``"border"``.
+
+            - NIfTI files {``"zeros"``, ``"border"``, ``"reflection"``}
+                Padding mode for outside grid values.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#grid-sample
+            - PNG files
+                This option is ignored.
+
+        scale: {``255``, ``65535``} postprocess data by clipping to [0, 1] and scaling
+            [0, 255] (uint8) or [0, 65535] (uint16). Default is None to disable scaling.
+            it's used for PNG format only.
+        dtype: data type during resampling computation. Defaults to ``np.float64`` for best precision.
+            if None, use the data type of input data. To be compatible with other modules,
+            the output data type is always ``np.float32``.
+            it's used for NIfTI format only.
+        output_dtype: data type for saving data. Defaults to ``np.float32``.
+            it's used for NIfTI format only.
+        save_batch: whether the import image is a batch data, default to `False`.
+            usually pre-transforms run for channel first data, while post-transforms run for batch data.
+
+    """
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        meta_key_postfix: str = "meta_dict",
+        output_dir: str = "./",
+        output_postfix: str = "trans",
+        output_ext: str = ".nii.gz",
+        resample: bool = True,
+        mode: Union[GridSampleMode, InterpolateMode, str] = "nearest",
+        padding_mode: Union[GridSamplePadMode, str] = GridSamplePadMode.BORDER,
+        scale: Optional[int] = None,
+        dtype: DtypeLike = np.float64,
+        output_dtype: DtypeLike = np.float32,
+        save_batch: bool = False,
+    ) -> None:
+        super().__init__(keys)
+        self.meta_key_postfix = meta_key_postfix
+        self._saver = SaveImage(
+            output_dir=output_dir,
+            output_postfix=output_postfix,
+            output_ext=output_ext,
+            resample=resample,
+            mode=mode,
+            padding_mode=padding_mode,
+            scale=scale,
+            dtype=dtype,
+            output_dtype=output_dtype,
+            save_batch=save_batch,
+        )
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            meta_data = d[f"{key}_{self.meta_key_postfix}"] if self.meta_key_postfix is not None else None
```

Comment:
if the image is loaded through numpy, can it still support saving as nifti files?

Reply:
I think it's should work, just set `meta_key_postfix=None`, you can have a try.

Thanks.

Reply:
can't we not force the user to pass `meta_key_postfix=None`.  That way, user can use LoadImaged for both Numpy/Nifti.  I guess somehow NumpyReader doesn't create the meta_data 

Instead you can change it to
`meta_data = d.get(f"{key}_{self.meta_key_postfix}") if self.meta_key_postfix is not None else None`

Reply:
Hi Sachi,

`NumpyReader` also creates meta_data:
https://github.com/Project-MONAI/MONAI/blob/master/monai/data/image_reader.py#L512
So I think we have meta_data by default.

Thanks for your suggestions.